.pre-commit-config.yaml

```diff
@@ -0,0 +1,59 @@
+# Pre-commit is tool which allows to run checks automatically before commit.
+# This way commits are always consistent and clean.
+
+exclude: target/|\.git/|.*\.snap
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: c4a0b883114b00d8d76b479c820ce7950211c99b # v4.5.0
+    hooks:
+      - id: check-merge-conflict
+        args: [--assume-in-merge]
+      # Editorconfig-checker does not care if file contains multiple newlines.
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-case-conflict
+      - id: trailing-whitespace
+      - id: fix-byte-order-marker
+
+  - repo: https://github.com/Lucas-C/pre-commit-hooks
```

Comment:
Could this be done in the rust formatter maybe? In combination with the .editorconfig it might be enough. 
https://rust-lang.github.io/rustfmt/?version=v1.7.0&search=#tab_spaces

Strangely, some of the files I saw affected were rust files. I wonder what the formatter does with raw strings in rust in that case.

Reply:
I think we cannot. This is mostly for other files than .rs. Also this affects rust files because comments can have tabs. Rust fmt should not touch comments and that is fine and good desing, but in this project imo we do not need tabs even in comments. We might need those in some test but then again we would just do `exclude:` for that particular test file in pre-commit-config.

Editorconfig itself does create job when doing editing but it does not prevent them happening. These checks makes devs happy as tabs are auto converted to spaces before commit.